### PR TITLE
Welding airlocks changes their layer

### PR DESCRIPTION
**Pull request by Elijahrane**
## About the PR 

**Media**

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

https://user-images.githubusercontent.com/60792108/211904335-202e78aa-1158-489e-b0d7-da67a2304ddc.mp4

**Changelog**

:cl: Rane
- tweak: Welding doors will now stop mice, rat kings and other mobs from running under them.

### CODE REVIEW

File: Content.Server/Doors/Systems/DoorSystem.cs [hunk 1]
```diff
@@ -148,9 +150,15 @@ private void OnWeldAttempt(EntityUid uid, DoorComponent component, WeldableAttem
...
     {
         if (component.State == DoorState.Closed)
+        {
             SetState(uid, DoorState.Welded, component);
+            AdjustLayer(uid, true);
+        }
         else if (component.State == DoorState.Welded)
+        {
             SetState(uid, DoorState.Closed, component);
+            AdjustLayer(uid, false);
```
Comment (mirrorcult) [inline Content.Server/Doors/Systems/DoorSystem.cs:160]: why is this changed in the door handler? `LayerChangeOnWeldComponent` should have its own subscription so it isnt tied to door. if this is needed because of some indirection with the door being open/closed then the layerchangeonweld can raise an event or the door system can remove the component when open or something

**Review by mirrorcult — CHANGES_REQUESTED** [review]

**Comment by Elijahrane** [comment]
you are right it is more broadly applicable than doors; I have moved everything over to the tools namespace

Currently mispredicts due to https://github.com/space-wizards/space-station-14/issues/13616 as a caution, probably want that fixed before merging

**Comment by Visne** [comment]
I've fixed merge conflicts for you
